### PR TITLE
DX-425: refactor nightwatch

### PR DESCRIPTION
**Pull request by JeremyCraigMartinez**

### CODE REVIEW

File: _test/browser/helpers/apiReference/methods/layout.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+exports.command = function navigateTo(NUMAPIS, expectedNumberOfApiEndpoints) {
```
Comment (willwhitwell) [inline _test/browser/helpers/apiReference/methods/layout.js]: Oddly named assertion...

File: _test/browser/nightwatch.json [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+    ],
     "output_folder": "_test/browser/_results",
-    "custom_commands_path": "",
-    "custom_assertions_path": "",
-    "page_objects_path": "",
-    "globals_path": "",
+    "custom_commands_path": "_test/browser/helpers",
+    "custom_assertions_path": "_test/browser/assertions",
+    "page_objects_path": "_test/browser/pages",
+    "globals_path": "_test/browser/globals.js",
```
Comment (willwhitwell) [inline _test/browser/nightwatch.json]: Very nice!

File: _test/browser/nightwatch.json [hunk 3]
```diff
@@ -21,10 +24,7 @@
                 "browserName": "chrome",
                 "javascriptEnabled": true,
                 "acceptSslCerts": true
-            },
-            "exclude": [
-                "helpers"
-            ]
+            }
```
Comment (willwhitwell) [inline _test/browser/nightwatch.json]: You can drop a "globals" object in "default" and set a "waitForConditionTimeout" (likely dependent on the version of Nightwatch) and then you have a global throughout your tests for any "waitFor*" command which is pretty nice.

File: _test/browser/pages/endpointSummary.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        'summary': '.endpoint-summary',
+        'requestConsole': '.console-req-container .code-snippet',
+        'responseConsole': '.console-res-container .code-snippet',
+        'responseConsole-GET': '.api-console-output .code-snippet',
+        'consoleText': 'span:first-of-type'
+    },
+
+    commands: [
+        {
+            getConsoleText: function(method, type, callback) {
```
Comment (willwhitwell) [inline _test/browser/pages/endpointSummary.js]: Pretty sweet! Would have to use this later!

It almost could be converted into a generic command or even an assertion based on your usage.

File: _test/browser/assertions/elementNumTimes.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+exports.assertion = function assertion(selector, expected, messageType) {
```
Comment (willwhitwell) [inline _test/browser/assertions/elementNumTimes.js]: Haven't written my own assertions yet and this looks interesting!

Did you want to name the function "assertion"??

File: _test/browser/api-reference/avatax.js [hunk 6]
```diff
@@ -27,123 +22,72 @@ module.exports = {
...
-        const expectedNumberOfApiEndpoints = 26;
+        expectedNumberOfApiEndpoints = 26;
         const expectedNumberOfSubTags = 5;
 
         browser
-            .url(this.baseURL + '/api-reference/avatax/rest/v2/methods/Accounts/AccountResetLicenseKey/')
-            .waitForElementVisible('[data-reactroot]', this.waitTime)
+            .url(browser.globals.baseURL + '/api-reference/avatax/rest/v2/methods/Accounts/AccountResetLicenseKey/')
+            .waitForElementVisible('[data-reactroot]', browser.globals.waitTime)
+            .apiReference.methods.layout(NUMAPIS, expectedNumberOfApiEndpoints + expectedNumberOfSubTags);
```
Comment (willwhitwell) [inline _test/browser/api-reference/avatax.js]: You could even drop this chain into a special assertion seeing as it is used over and over again slimming this down even further!

File: _test/browser/api-reference/avatax.js [hunk 7]
```diff
@@ -1,18 +1,13 @@
...
-    'baseURL': process.env.BASEURL ? process.env.BASEURL.replace(/\/$/, '') : 'http://localhost:4000',
-    'waitTime': isNaN(parseInt(process.env.TIMEOUT, 10)) ? 5000 : parseInt(process.env.TIMEOUT, 10),
     'before': function(browser) {
-        /* eslint-disable no-console */
-        console.log('WaitTime set to', this.waitTime);
-        console.log('BaseURL set to', this.baseURL);
-        /* eslint-enable no-console */
-
         browser.maximizeWindow();
+        navigationBar = browser.page.navigationBar();
```
Comment (willwhitwell) [inline _test/browser/api-reference/avatax.js]: Nice! - this is how we accomplished this initialization as well!

File: _test/browser/assertions/elementNumTimes.js (hunk 5 above)
Comment (JeremyCraigMartinez) [inline _test/browser/assertions/elementNumTimes.js]: I'm not sure if the name of this function would appear on the callstack. But if so no, I'll change it either way

**Comment by JeremyCraigMartinez** [comment]
Thanks for the tips @willwhitwell! Code looks much cleaner now!

**Pull request merged**